pkg/util/pki/basicconstraints.go

```diff
@@ -28,14 +29,40 @@ var (
 
 // Copied from x509.go
 type basicConstraints struct {
-	IsCA bool
+	IsCA       bool `asn1:"optional"`
+	MaxPathLen int  `asn1:"optional,default:-1"`
 }
 
 // Adapted from x509.go
-func MarshalBasicConstraints(isCA bool) (pkix.Extension, error) {
+func MarshalBasicConstraints(isCA bool, maxPathLen *int) (pkix.Extension, error) {
```

Comment:
suggestion: Only highlighting this because you explicitly said in slack that the PR "should not contain any breaking changes anymore" - this is a breaking change.

IMO this is probably an OK one since there's basically no chance we'll ever have to change the subject again after this and I suspect this function is very unlikely to be used externally. I also wouldn't mind us doing `MarshalBasicConstraintsWithPathLen` and having `MarshalBasicConstraints` just call that new function, though.

Reply:
Ah, right I missed this function.